proxy/proxy_intercept.go

```diff
@@ -78,17 +78,14 @@ func doRawStream(w http.ResponseWriter, resp *http.Response, client *httputil.Cl
 	defer down.Close()
 	defer up.Close()
 
-	if _, err := down.Write([]byte("HTTP/1.1 200 OK\n")); err != nil {
-		Log.Warning(err)
-		return
-	}
-
-	if err := resp.Header.Write(down); err != nil {
-		Log.Warning(err)
-		return
+	if _, err = downBuf.Write([]byte("HTTP/1.1 " + resp.Status + "\n")); err == nil {
+		if err = resp.Header.Write(downBuf); err == nil {
+			if _, err = downBuf.Write([]byte("\n")); err == nil {
+				err = downBuf.Flush()
+			}
+		}
```

Comment:
This nesting for four steps is horrible.

I suggest flattening it and using a 'defer' to log warnings,